src/pytorch_lightning/utilities/types.py

```diff
@@ -101,6 +101,18 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
         ...
 
 
+@runtime_checkable
+class _IntStateful(Protocol):
```

Comment:
I think this could be done with Generics where `_Stateful` takes a `Generic[T]` which here is `int`

Reply:
Alright, let us try modifying `_Stateful`.

Reply:
The following modification passes a `mypy` check.

https://github.com/Lightning-AI/lightning/blob/c11d3e03ea8923f25d098478328ab60da6e9a325/src/pytorch_lightning/utilities/types.py#L96-L102

Reply:
Thanks! Pushed https://github.com/Lightning-AI/lightning/pull/13904/commits/71d81da6d870b16b526297ee05d658affd44fe7c to finish it